subsys/bluetooth/controller/hal/nrf5/radio/radio_nrf52810.h

```diff
@@ -0,0 +1,358 @@
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_1M_DEFAULT_NS
+
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_US \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_DEFAULT_US
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_NS \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_DEFAULT_NS
+
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_US \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_DEFAULT_US
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_NS \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_DEFAULT_NS
+
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_US \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_DEFAULT_US
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_NS \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_DEFAULT_NS
+
+#else /* !CONFIG_BT_CTLR_TIFS_HW */
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_1M_US \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_1M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_1M_NS \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_1M_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_US \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_NS \
+	HAL_RADIO_NRF52810_TXEN_TXIDLE_TX_2M_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_US \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_NS \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_1M_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_US \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_NS \
+	HAL_RADIO_NRF52810_RXEN_RXIDLE_RX_2M_DEFAULT_NO_HW_TIFS_NS
+#endif /* !CONFIG_BT_CTLR_TIFS_HW */
+#endif /* !CONFIG_BT_CTLR_RADIO_ENABLE_FAST */
+
+#if !defined(CONFIG_BT_CTLR_TIFS_HW)
+#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+#undef EVENT_TIMER
+#define EVENT_TIMER NRF_TIMER4
+#define SW_SWITCH_TIMER EVENT_TIMER
+#define SW_SWITCH_TIMER_EVTS_COMP_BASE 4
+#else /* !CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER */
+#define SW_SWITCH_TIMER NRF_TIMER1
+#define SW_SWITCH_TIMER_EVTS_COMP_BASE 0
+#endif /* !CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER */
+
+#define SW_SWITCH_TIMER_TASK_GROUP_BASE 0
+#endif /* !CONFIG_BT_CTLR_TIFS_HW */
+
+static inline void hal_radio_reset(void)
+{
+}
+
+static inline void hal_radio_ram_prio_setup(void)
+{
+#if !defined(CONFIG_SOC_SERIES_NWTSIM_NRFXX)
+	struct {
+		u32_t volatile reserved_0[0x5a0 >> 2];
+		u32_t volatile bridge_type;
+		u32_t volatile reserved_1[((0xe00 - 0x5a0) >> 2) - 1];
+		struct {
+			u32_t volatile CPU0;
+			u32_t volatile SPIS1;
+			u32_t volatile RADIO;
+			u32_t volatile ECB;
+			u32_t volatile CCM;
+			u32_t volatile AAR;
+			u32_t volatile SAADC;
+			u32_t volatile UARTE;
+			u32_t volatile SERIAL0;
+			u32_t volatile SERIAL2;
+			u32_t volatile NFCT;
+			u32_t volatile I2S;
+			u32_t volatile PDM;
+			u32_t volatile PWM;
+		} RAMPRI;
+	} volatile *NRF_AMLI = (void volatile *)0x40000000UL;
+
+	NRF_AMLI->RAMPRI.CPU0    = 0xFFFFFFFFUL;
```

Comment:
Can we use the nrfx radio hal for this? If not then I think we should add it in an upcoming version of nrfx, @anangl?

Reply:
No, the nrfx RADIO HAL is only for accessing the RADIO peripheral.

Reply:
OK, thanks. In that case can you make a note to provide nrfx hal functions for AMLI in the future?